os/arch/arm/src/armv7-m/up_busfault.c

```diff
@@ -0,0 +1,144 @@
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+
+#include <assert.h>
+#include <debug.h>
+
+#include <arch/irq.h>
+
+#include "up_arch.h"
+#include "nvic.h"
+#include "up_internal.h"
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define MASK_BFARVALID		0x00008000
+#define MASK_LSPERR		0x00002000
+#define MASK_STKERR		0x00001000
+#define MASK_UNSTKERR		0x00000800
+#define MASK_IMPRECISERR	0x00000400
+#define MASK_PRECISERR		0x00000200
+#define MASK_IBUSERR		0x00000100
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: up_busfault
+ *
+ * Description:
+ *   This is Bus Fault exception handler.
+ *
+ ****************************************************************************/
+
+int up_busfault(int irq, FAR void *context, FAR void *arg)
+{
+	(void)irqsave();
+	uint32_t *regs = (uint32_t *)context;
+
+	uint32_t cfsr = getreg32(NVIC_CFAULTS);
+	uint32_t bfar = getreg32(NVIC_BFAULT_ADDR);
+
+	dbg("PANIC!!! Bus fault occured while executing instruction at address : 0x%08x\n", regs[REG_R15]);
+	lldbg("CFAULTS: 0x%08x BFAR: 0x%08x\n", cfsr, bfar);
+
+	if(cfsr & MASK_PRECISERR){
+		lldbg("Precise data access error occured.\n");
+	}
+	else if(cfsr & MASK_IMPRECISERR){
+		lldbg("Imprecise data access error occured.\n");
+	}
+	else if(cfsr & MASK_STKERR){
+		lldbg("Busfault occurred while stacking registers during exception entry.\n");
```

Comment:
I think we can know which fault is occurred from line 112. So we can reduce the description here, just
"while stacking registers during exception entry."